api/app/helpers/spree/api/api_helpers.rb

```diff
@@ -121,11 +121,18 @@ def required_fields_for(model)
         :id, :number, :state, :order_id, :memo, :created_at, :updated_at
       ]
 
-      @@address_attributes = [
-        :id, :firstname, :lastname, :full_name, :address1, :address2, :city,
-        :zipcode, :phone, :company, :alternative_phone, :country_id, :country_iso,
-        :state_id, :state_name, :state_text
-      ]
+      @@full_address_attributes = [
+        :id, :name, :firstname, :lastname, :full_name, :address1, :address2,
+        :city, :zipcode, :phone, :company, :alternative_phone, :country_id,
+        :country_iso, :state_id, :state_name, :state_text
+      ]
+
+      @@address_attributes = if Spree::Config[:show_address_deprecated_attributes]
+                               @@full_address_attributes
+                             else
+                               @@full_address_attributes -
+                                 Spree::Address::DEPRECATED_ATTRS.map(&:to_sym)
```

Comment:
If these are deprecated attributes should the logic here be additive instead? This makes the default list the non-deprecated fields with deprecated fields being added.

For example,

```
@@address_attributes = if Spree::Config[:show_address_deprecated_attributes]
                                               @@full_address_attributes + 
                                                 Spree::Address::DEPRECATED_ATTRS.map(&:to_sym)
                                             else
                                               @@full_address_attributes
                                              end
```

Reply:
Yes, it would me more readable, thank you! 👍